Disclaimer - Use results and my commentary at your own risk - I am not a medical professional or a health statistician.

This is just me playing with some data and creating stories from it.  I hope you enjoy and learn something.

All data from https://covidtracking.com/data

Data ran in this notebook is current as of June 1, 2020.

Create a local path to download the csv data to

In [4]:
localpath="dbfs:/USACovid19/"

##Remove directory
dbutils.fs.rm(localpath,True)

Out[21]: True

In [5]:
dbutils.fs.mkdirs(localpath)



Out[22]: True

Download COVID19 data

In [7]:
%sh
wget -P /dbfs/USACovid19/ https://covidtracking.com/api/v1/states/daily.csv

--2020-06-02 21:15:52-- https://covidtracking.com/api/v1/states/daily.csv
Resolving covidtracking.com (covidtracking.com)... 162.243.166.170, 2604:a880:400:d1::888:7001
Connecting to covidtracking.com (covidtracking.com)|162.243.166.170|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 986984 (964K) [text/csv]
Saving to: ‘/dbfs/USACovid19/daily.csv’

 0K .......... .......... .......... .......... .......... 5% 3.61M 0s
 50K .......... .......... .......... .......... .......... 10% 23.0M 0s
 100K .......... .......... .......... .......... .......... 15% 29.1M 0s
 150K .......... .......... .......... .......... .......... 20% 27.8M 0s
 200K .......... .......... .......... .......... .......... 25% 15.2M 0s
 250K .......... .......... .......... .......... .......... 31% 7.66M 0s
 300K .......... .......... .......... .......... .......... 36% 8.75M 0s
 350K .......... .......... .......... .......... .......... 41% 8.41M 0s
 400K .......... .......... .......... .......... .......... 46% 8.13M 0s
 450K .......... .......... .......... .......... .......... 51% 22.6M 0s
 500K .......... .......... .......... .......... .......... 57% 8.66M 0s
 550K .......... .......... .......... .......... .......... 62% 8.33M 0s
 600K .......... .......... .......... .......... .......... 67% 7.19M 0s
 650K .......... .......... .......... .......... .......... 72% 25.6M 0s
 700K .......... .......... .......... .......... .......... 77% 8.77M 0s
 750K .......... .......... .......... .......... .......... 83% 7.17M 0s
 800K .......... .......... .......... .......... .......... 88% 7.47M 0s
 850K .......... .......... .......... .......... .......... 93% 8.35M 0s
 900K .......... .......... .......... .......... .......... 98% 14.8M 0s
 950K .......... ... 100% 25.0M=0.1s

2020-06-02 21:15:52 (9.70 MB/s) - ‘/dbfs/USACovid19/daily.csv’ saved [986984/986984]

In [8]:
dbutils.fs.ls ('dbfs:/USACovid19')

Out[24]: [FileInfo(path='dbfs:/USACovid19/daily.csv', name='daily.csv', size=986984)]

Create a Dataframe to hold COVID19 data for all US states reporting

In [10]:
dfUSACovid19 = (
  spark
  .read
  .option ("header",True)
  .option ("inferschema",True)
  .csv("dbfs:/USACovid19/daily.csv")
      )

In [11]:
display(dfUSACovid19).take(20)

date state positive negative pending hospitalizedCurrently hospitalizedCumulative inIcuCurrently inIcuCumulative onVentilatorCurrently onVentilatorCumulative recovered dataQualityGrade lastUpdateEt dateModified checkTimeEt death hospitalized dateChecked fips positiveIncrease negativeIncrease total totalTestResults totalTestResultsIncrease posNeg deathIncrease hospitalizedIncrease hash commercialScore negativeRegularScore negativeScore positiveScore score grade 20200602 AK 487 55716 null 10 null null null 0.0 null 371 A 6/2/2020 00:00 2020-06-02T00:00:00.000+0000 06/01 20:00 10 null 2020-06-02T00:00:00.000+0000 2 20 1993 56203 56203 2013 56203 0 0 8bfc576f3ce0c5438919ab2a672255d0630714ec 0 0 0 0 0 null 20200602 AL 18642 205160 null null 1879 null 591 null 355 9355 B 6/1/2020 00:00 2020-06-01T00:00:00.000+0000 05/31 20:00 651 1879 2020-06-01T00:00:00.000+0000 1 279 0 223802 223802 279 223802 5 23 e925d317dd6bba774728abaf5db247f3c5983394 0 0 0 0 0 null 20200602 AR 7443 125793 null 121 718 null null 26.0 124 5401 A 6/2/2020 00:00 2020-06-02T00:00:00.000+0000 06/01 20:00 133 718 2020-06-02T00:00:00.000+0000 5 0 0 133236 133236 0 133236 0 7 1456f854de8dbd854d74cf88314b9c36bdf5c78a 0 0 0 0 0 null 20200602 AS 0 174 null null null null null null null null C 5/29/2020 00:00 2020-05-29T00:00:00.000+0000 05/28 20:00 0 null 2020-05-29T00:00:00.000+0000 60 0 0 174 174 0 174 0 0 cccae3dfdebc5c6a2b53fdb498e0fbce6aec0120 0 0 0 0 0 null 20200602 AZ 21250 216583 null 1009 3057 380 null 235.0 null 4967 A+ 6/2/2020 00:00 2020-06-02T00:00:00.000+0000 06/01 20:00 941 3057 2020-06-02T00:00:00.000+0000 4 1127 8636 237833 237833 9763 237833 24 39 40c3b8895be0300a58790c838cb9a3a154d62cc0 0 0 0 0 0 null 20200602 CA 115310 1956281 null 4393 null 1286 null null null null B 6/2/2020 00:00 2020-06-02T00:00:00.000+0000 06/01 20:00 4286 null 2020-06-02T00:00:00.000+0000 6 2304 56704 2071591 2071591 59008 2071591 35 0 9c03c1926787907836d6c35b13c8e44c97dc26b7 0 0 0 0 0 null 20200602 CO 26577 163960 null 394 4372 null null null null 3954 A 6/2/2020 00:00 2020-06-02T00:00:00.000+0000 06/01 20:00 1458 4372 2020-06-02T00:00:00.000+0000 8 199 3962 190537 190537 4161 190537 13 25 0a4c8534ed6ced43b0835e9e7da33d18914d4141 0 0 0 0 0 null 20200602 CT 42979 227631 null 434 13793 null null null null 7511 B 6/1/2020 20:30 2020-06-01T20:30:00.000+0000 06/01 16:30 3972 13793 2020-06-01T20:30:00.000+0000 9 239 8981 270610 270610 9220 270610 8 0 d21862626f593d49b27bfa13773b02a3eebcd900 0 0 0 0 0 null 20200602 DC 8886 38815 null 295 null 106 null 74.0 null 1137 A+ 6/1/2020 00:00 2020-06-01T00:00:00.000+0000 05/31 20:00 470 null 2020-06-01T00:00:00.000+0000 11 29 409 47701 47701 438 47701 2 0 36f51c85e83d22ab21e78b35a025b3314b2be260 0 0 0 0 0 null 20200602 DE 9685 54377 null 157 null null null null null 5442 A 6/1/2020 18:00 2020-06-01T18:00:00.000+0000 06/01 14:00 373 null 2020-06-01T18:00:00.000+0000 10 80 1535 64062 64062 1615 64062 5 0 a6601e4f641beae62fccabfc1cf8aa554f65f260 0 0 0 0 0 null 20200602 FL 57447 992305 1206 null 10678 null null null null null A 6/1/2020 09:54 2020-06-01T09:54:00.000+0000 06/01 05:54 2613 10678 2020-06-01T09:54:00.000+0000 12 617 8735 1050958 1049752 9352 1049752 70 184 74a707be20d238b7e16ef3b6072ad4e5ecf7c170 0 0 0 0 0 null 20200602 GA 48207 426080 null 828 8334 null 1821 null null null A 6/2/2020 15:11 2020-06-02T15:11:00.000+0000 06/02 11:11 2102 8334 2020-06-02T15:11:00.000+0000 13 589 2067 474287 474287 2656 474287 28 207 cd795b1de96ac70cfa52c73b6260c76a598cb509 0 0 0 0 0 null 20200602 GU 177 6214 null null null null null null null 148 B 6/2/2020 06:15 2020-06-02T06:15:00.000+0000 06/02 02:15 5 null 2020-06-02T06:15:00.000+0000 66 2 73 6391 6391 75 6391 0 0 d55ffe7e8404632c76a26193e66abf4a758a8a64 0 0 0 0 0 null 20200602 HI 652 48269 null null 83 null null null null 608 D 6/1/2020 18:00 2020-06-01T18:00:00.000+0000 06/01 14:00 17 83 2020-06-01T18:00:00.000+0000 15 0 434 48921 48921 434 48921 0 0 db7cae23fab4e5c378691131d021b311c6335023 0 0 0 0

For my story, I chose to play with State of Maryland Data

Creating a new dataframe for only MD data from March 29, 2020

In [14]:
##State of Maryland didn't report negatives between 3/12 and 3/28 so data will be used starting 3/29/2020

from pyspark.sql import *

dfMDCovid19 = dfUSACovid19.filter((dfUSACovid19.state == "MD") & (dfUSACovid19.date >= "20200329")).sort(dfUSACovid19.date)

display(dfMDCovid19)

date state positive negative pending hospitalizedCurrently hospitalizedCumulative inIcuCurrently inIcuCumulative onVentilatorCurrently onVentilatorCumulative recovered dataQualityGrade lastUpdateEt dateModified checkTimeEt death hospitalized dateChecked fips positiveIncrease negativeIncrease total totalTestResults totalTestResultsIncrease posNeg deathIncrease hospitalizedIncrease hash commercialScore negativeRegularScore negativeScore positiveScore score grade 20200329 MD 1239 12354 null null 277 null null null null null D 3/29/2020 10:00 2020-03-29T10:00:00.000+0000 03/29 06:00 10 277 2020-03-29T10:00:00.000+0000 24 247 838 13593 13593 1085 13593 5 51 99c2a57a9870ca53eb8e08e49afad92542cb62e5 0 0 0 0 0 null 20200330 MD 1413 13316 null null 353 null null null null null A 3/30/2020 10:00 2020-03-30T10:00:00.000+0000 03/30 06:00 15 353 2020-03-30T10:00:00.000+0000 24 174 962 14729 14729 1136 14729 5 76 5666d5f891c2c7bb73cb33020917d93fd2706b6e 0 0 0 0 0 null 20200331 MD 1660 14868 null null 429 null null null null null A 3/31/2020 10:00 2020-03-31T10:00:00.000+0000 03/31 06:00 18 429 2020-03-31T10:00:00.000+0000 24 247 1552 16528 16528 1799 16528 3 76 d6eaae443a0b9bf17605c9f8b91cde22f1525ea7 0 0 0 0 0 null 20200401 MD 1985 17233 null null 522 null null null null null A 4/1/2020 10:00 2020-04-01T10:00:00.000+0000 04/01 06:00 31 522 2020-04-01T10:00:00.000+0000 24 325 2365 19218 19218 2690 19218 13 93 dc24f941a30f7b77f2c42b02918167150bedc6df 0 0 0 0 0 null 20200402 MD 2331 18890 null null 582 null null null null 81 A 4/2/2020 10:00 2020-04-02T10:00:00.000+0000 04/02 06:00 36 582 2020-04-02T10:00:00.000+0000 24 346 1657 21221 21221 2003 21221 5 60 2524161fea18dbe78f40d2f095c9dc1a5b515637 0 0 0 0 0 null 20200403 MD 2758 20932 null null 664 null null null null 159 A 4/3/2020 10:00 2020-04-03T10:00:00.000+0000 04/03 06:00 42 664 2020-04-03T10:00:00.000+0000 24 427 2042 23690 23690 2469 23690 6 82 7346ddbf0fee002be1916f6e12661ce264310c3d 0 0 0 0 0 null 20200404 MD 3125 22485 null null 821 null null null null 159 A 4/4/2020 10:00 2020-04-04T10:00:00.000+0000 04/04 06:00 53 821 2020-04-04T10:00:00.000+0000 24 367 1553 25610 25610 1920 25610 11 157 7b5bd8eb0e76c1abb2776ce773247a13f870f250 0 0 0 0 0 null 20200405 MD 3609 24728 null null 936 null null null null 159 A 4/5/2020 10:00 2020-04-05T10:00:00.000+0000 04/05 06:00 67 936 2020-04-05T10:00:00.000+0000 24 484 2243 28337 28337 2727 28337 14 115 636737bab78b465c34b5f9f3ef456759243031f2 0 0 0 0 0 null 20200406 MD 4045 25572 null null 1059 null null null null 184 A 4/6/2020 10:00 2020-04-06T10:00:00.000+0000 04/06 06:00 91 1059 2020-04-06T10:00:00.000+0000 24 436 844 29617 29617 1280 29617 24 123 f6ce757f3ddee0d306384d868a1a0fc678e02c05 0 0 0 0 0 null 20200407 MD 4371 27256 null null 1106 null null null null 288 A 4/7/2020 10:00 2020-04-07T10:00:00.000+0000 04/07 06:00 103 1106 2020-04-07T10:00:00.000+0000 24 326 1684 31627 31627 2010 31627 12 47 1e64810c79e672682e5a8e0b7160310e3261aa76 0 0 0 0 0 null 20200408 MD 5529 32933 null null 1210 null null null null 365 A 4/8/2020 10:00 2020-04-08T10:00:00.000+0000 04/08 06:00 124 1210 2020-04-08T10:00:00.000+0000 24 1158 5677 38462 38462 6835 38462 21 104 22ba44366f4f4b12d08c5cfd29b337ade901bf76 0 0 0 0 0 null 20200409 MD 6185 35344 null null 1348 null null null null 376 A 4/9/2020 10:00 2020-04-09T10:00:00.000+0000 04/09 06:00 138 1348 2020-04-09T10:00:00.000+0000 24 656 2411 41529 41529 3067 41529 14 138 1b0c56678462dbdf779ad7fabb3bcb557179add1 0 0 0 0 0 null 20200410 MD 6968 37480 null null 1413 null null null null 397 A 4/10/2020 10:00 2020-04-10T10:00:00.000+0000 04/10 06:00 171 1413 2020-04-10T10:00:00.000+0000 24 783 2136 44448 44448 2919 44448 33 65 ea8233a5be17c452ed9c99d0842ddd090751c7b9 0 0 0 0 0 null 20200411 MD 7694 39544 null null 1709 null null null null 431 A 4/11/2020 10:00 2020-04-11T10:00:00.000+0000 04/11 06:00 206 1709 2020-04-11T10:00:00.000+0000 24 726 2064 47238 47238 2790 47238 35 296 dfa5597ec582a5c932c8e77fc7a03d1f8c086b6

Creating a view from the Dataframe so I can run SQL queries against it

In [16]:
dfMDCovid19.createOrReplaceTempView("vw_SQLMDCovid19")

Ordered by the date with the highest percent positive rating (for cases) descending

In [18]:
%sql
Select date,totalTestResultsIncrease,positiveIncrease,negativeIncrease, 
positiveIncrease/totalTestResultsIncrease*100 as PercentPositive, negativeIncrease/totalTestResultsIncrease*100 as PercentNegative
from vw_SQLMDCovid19
order by PercentPositive desc

date totalTestResultsIncrease positiveIncrease negativeIncrease PercentPositive PercentNegative 20200413 1987 711 1276 35.782586814292905 64.2174131857071 20200406 1280 436 844 34.0625 65.9375 20200508 3587 1111 2476 30.972957903540564 69.02704209645944 20200415 2030 560 1470 27.586206896551722 72.41379310344827 20200509 3841 1049 2792 27.310596198906534 72.68940380109346 20200414 1982 536 1446 27.04339051463169 72.95660948536832 20200510 3903 1053 2850 26.979246733282093 73.0207532667179 20200423 3571 962 2609 26.939232707924948 73.06076729207504 20200410 2919 783 2136 26.82425488180884 73.17574511819116 20200506 3909 1046 2863 26.7587618316705 73.24123816832949 20200507 4588 1211 3377 26.3949433304272 73.6050566695728 20200501 6624 1730 4894 26.117149758454104 73.8828502415459 20200425 4407 1150 3257 26.094849103698657 73.90515089630134 20200430 3430 893 2537 26.034985422740526 73.96501457725948 20200411 2790 726 2064 26.021505376344084 73.9784946236559 20200519 7152 1784 5368 24.94407158836689 75.0559284116331 20200417 3166 788 2378 24.88945041061276 75.11054958938725 20200515 4372 1083 3289 24.771271729185727 75.22872827081427 20200521 4903 1208 3695 24.637976748929226 75.36202325107078 20200416 3080 752 2328 24.415584415584416 75.58441558441558 20200420 3506 854 2652 24.358243011979464 75.64175698802053 20200523 4425 1071 3354 24.203389830508474 75.79661016949153 20200429 3144 736 2408 23.40966921119593 76.59033078880407 20200505 3108 709 2399 22.81209781209781 77.1879021879022 20200329 1085 247 838 22.764976958525345 77.23502304147466 20200421 2238 509 1729 22.743521000893654 77.25647899910635 20200504 4201 946 3255 22.51844798857415 77.48155201142585 20200514 4880 1091 3789 22.356557377049178 77.64344262295081 20200428 2809 626 2183 22.285510857956567 77.71448914204343 20200502 4539 1001 3538 22.053315708305792 77.9466842916942 20200418 3361 736 2625 21.89824457006843 78.10175542993156 20200409 3067 656 2411 21.388979458754484 78.61102054124551 20200412 2526 531 1995 21.021377672209027 78.97862232779097 20200419 2521 522 1999 20.706069020230068 79.29393097976994 20200422 2894 582 2312 20.11057360055287 79.88942639944713 20200404 1920 367 1553 19.114583333333332 80.88541666666667 20200424 4616 879 3737 19.042461005199307 80.9575389948007 20200513 4071 751 3320 18.447555883075413 81.55244411692459 20200405 2727 484 2243 17.74844151081775 82.25155848918226 20200517 4751 836 3915 17.596295516733317 82.40370448326668 20200403 2469 427 2042 17.294451194815714 82.70554880518428 20200402 2003 346 1657 17.274088866699948 82.72591113330004 20200408 6835 1158 5677 16.942209217264082 83.05779078273592 20200407 2010 326 1684 16.218905472636816 83.78109452736318 20200511 4849 786 4063 16.209527737677874 83.79047226232213 20200330 1136 174 962 15.316901408450704 84.6830985915493 20200512 4723 688 4035 14.56701249206013 85.43298750793987 20200518 6580 958 5622 14.559270516717326 85.44072948328267 20200331 1799 247 1552 13.729849916620346 86.27015008337966 20200503 7272 989 6283 13.600110011001101 86.39988998899891 20200516 7349 982 6367 13.362362226153218 86.63763777384679 20200529 9660 1279 8381 13.24016563146998 86.75983436853002 20200401 2690 325 2365 12.0817843866171 87.9182156133829 20200602 7085 848 6237 11.968948482709951 88.03105151729005 20200520 6672 777 5895 11.64568345323741 88.35431654676259 20200522 7669 893 6776 11.64428217499022 88.35571782500978 20200427 8311 906 7405 10.901215256888461 89.09878474311154 20200526 4910 535 4375 10.896130346232178 89.10386965376782 20200530 9428 1027 8401 10.893084429359355 89.10691557064064 20200426 7542 815 6727 10.806152214266774 89.19384778573323 20200524 8035 818 7217 10.180460485376477 89.81953951462353 20200528 12803 1286 11517 10.044520815433883 89.95547918456612 20200527 7568 736 6832 9.725158562367865 90.27484143763213 20200531 7935 763 7172 9.615626969124133 90.38437303087586 20200525 9215 839 8376 9.104720564297342 90.89527943570266 20200601 6849 549 6300 8.015768725361367 91

Ordered by number of Positive Cases

In [20]:
%sql
Select date,totalTestResultsIncrease,positiveIncrease,negativeIncrease, 
positiveIncrease/totalTestResultsIncrease*100 as PercentPositive, negativeIncrease/totalTestResultsIncrease*100 as PercentNegative
from vw_SQLMDCovid19
order by positiveIncrease desc

date totalTestResultsIncrease positiveIncrease negativeIncrease PercentPositive PercentNegative 20200519 7152 1784 5368 24.94407158836689 75.0559284116331 20200501 6624 1730 4894 26.117149758454104 73.8828502415459 20200528 12803 1286 11517 10.044520815433883 89.95547918456612 20200529 9660 1279 8381 13.24016563146998 86.75983436853002 20200507 4588 1211 3377 26.3949433304272 73.6050566695728 20200521 4903 1208 3695 24.637976748929226 75.36202325107078 20200408 6835 1158 5677 16.942209217264082 83.05779078273592 20200425 4407 1150 3257 26.094849103698657 73.90515089630134 20200508 3587 1111 2476 30.972957903540564 69.02704209645944 20200514 4880 1091 3789 22.356557377049178 77.64344262295081 20200515 4372 1083 3289 24.771271729185727 75.22872827081427 20200523 4425 1071 3354 24.203389830508474 75.79661016949153 20200510 3903 1053 2850 26.979246733282093 73.0207532667179 20200509 3841 1049 2792 27.310596198906534 72.68940380109346 20200506 3909 1046 2863 26.7587618316705 73.24123816832949 20200530 9428 1027 8401 10.893084429359355 89.10691557064064 20200502 4539 1001 3538 22.053315708305792 77.9466842916942 20200503 7272 989 6283 13.600110011001101 86.39988998899891 20200516 7349 982 6367 13.362362226153218 86.63763777384679 20200423 3571 962 2609 26.939232707924948 73.06076729207504 20200518 6580 958 5622 14.559270516717326 85.44072948328267 20200504 4201 946 3255 22.51844798857415 77.48155201142585 20200427 8311 906 7405 10.901215256888461 89.09878474311154 20200522 7669 893 6776 11.64428217499022 88.35571782500978 20200430 3430 893 2537 26.034985422740526 73.96501457725948 20200424 4616 879 3737 19.042461005199307 80.9575389948007 20200420 3506 854 2652 24.358243011979464 75.64175698802053 20200602 7085 848 6237 11.968948482709951 88.03105151729005 20200525 9215 839 8376 9.104720564297342 90.89527943570266 20200517 4751 836 3915 17.596295516733317 82.40370448326668 20200524 8035 818 7217 10.180460485376477 89.81953951462353 20200426 7542 815 6727 10.806152214266774 89.19384778573323 20200417 3166 788 2378 24.88945041061276 75.11054958938725 20200511 4849 786 4063 16.209527737677874 83.79047226232213 20200410 2919 783 2136 26.82425488180884 73.17574511819116 20200520 6672 777 5895 11.64568345323741 88.35431654676259 20200531 7935 763 7172 9.615626969124133 90.38437303087586 20200416 3080 752 2328 24.415584415584416 75.58441558441558 20200513 4071 751 3320 18.447555883075413 81.55244411692459 20200418 3361 736 2625 21.89824457006843 78.10175542993156 20200429 3144 736 2408 23.40966921119593 76.59033078880407 20200527 7568 736 6832 9.725158562367865 90.27484143763213 20200411 2790 726 2064 26.021505376344084 73.9784946236559 20200413 1987 711 1276 35.782586814292905 64.2174131857071 20200505 3108 709 2399 22.81209781209781 77.1879021879022 20200512 4723 688 4035 14.56701249206013 85.43298750793987 20200409 3067 656 2411 21.388979458754484 78.61102054124551 20200428 2809 626 2183 22.285510857956567 77.71448914204343 20200422 2894 582 2312 20.11057360055287 79.88942639944713 20200415 2030 560 1470 27.586206896551722 72.41379310344827 20200601 6849 549 6300 8.015768725361367 91.98423127463863 20200414 1982 536 1446 27.04339051463169 72.95660948536832 20200526 4910 535 4375 10.896130346232178 89.10386965376782 20200412 2526 531 1995 21.021377672209027 78.97862232779097 20200419 2521 522 1999 20.706069020230068 79.29393097976994 20200421 2238 509 1729 22.743521000893654 77.25647899910635 20200405 2727 484 2243 17.74844151081775 82.25155848918226 20200406 1280 436 844 34.0625 65.9375 20200403 2469 427 2042 17.294451194815714 82.70554880518428 20200404 1920 367 1553 19.114583333333332 80.88541666666667 20200402 2003 346 1657 17.274088866699948 82.72591113330004 20200407 2010 326 1684 16.218905472636816 83.78109452736318 20200401 2690 325 2365 12.0817843866171 87.9182156133829 20200331 1799 247 1552 13.729849916620346 86.27015008337966 20200329 1085 247 838 22.764976958525345 77.23502304147466 20200330 1136 174 962 15.316901408450704 8

Ordered by Date (latest first)

In [22]:
%sql
Select date,totalTestResultsIncrease,positiveIncrease,negativeIncrease, 
positiveIncrease/totalTestResultsIncrease*100 as PercentPositive, negativeIncrease/totalTestResultsIncrease*100 as PercentNegative
from vw_SQLMDCovid19
order by date desc

date totalTestResultsIncrease positiveIncrease negativeIncrease PercentPositive PercentNegative 20200602 7085 848 6237 11.968948482709951 88.03105151729005 20200601 6849 549 6300 8.015768725361367 91.98423127463863 20200531 7935 763 7172 9.615626969124133 90.38437303087586 20200530 9428 1027 8401 10.893084429359355 89.10691557064064 20200529 9660 1279 8381 13.24016563146998 86.75983436853002 20200528 12803 1286 11517 10.044520815433883 89.95547918456612 20200527 7568 736 6832 9.725158562367865 90.27484143763213 20200526 4910 535 4375 10.896130346232178 89.10386965376782 20200525 9215 839 8376 9.104720564297342 90.89527943570266 20200524 8035 818 7217 10.180460485376477 89.81953951462353 20200523 4425 1071 3354 24.203389830508474 75.79661016949153 20200522 7669 893 6776 11.64428217499022 88.35571782500978 20200521 4903 1208 3695 24.637976748929226 75.36202325107078 20200520 6672 777 5895 11.64568345323741 88.35431654676259 20200519 7152 1784 5368 24.94407158836689 75.0559284116331 20200518 6580 958 5622 14.559270516717326 85.44072948328267 20200517 4751 836 3915 17.596295516733317 82.40370448326668 20200516 7349 982 6367 13.362362226153218 86.63763777384679 20200515 4372 1083 3289 24.771271729185727 75.22872827081427 20200514 4880 1091 3789 22.356557377049178 77.64344262295081 20200513 4071 751 3320 18.447555883075413 81.55244411692459 20200512 4723 688 4035 14.56701249206013 85.43298750793987 20200511 4849 786 4063 16.209527737677874 83.79047226232213 20200510 3903 1053 2850 26.979246733282093 73.0207532667179 20200509 3841 1049 2792 27.310596198906534 72.68940380109346 20200508 3587 1111 2476 30.972957903540564 69.02704209645944 20200507 4588 1211 3377 26.3949433304272 73.6050566695728 20200506 3909 1046 2863 26.7587618316705 73.24123816832949 20200505 3108 709 2399 22.81209781209781 77.1879021879022 20200504 4201 946 3255 22.51844798857415 77.48155201142585 20200503 7272 989 6283 13.600110011001101 86.39988998899891 20200502 4539 1001 3538 22.053315708305792 77.9466842916942 20200501 6624 1730 4894 26.117149758454104 73.8828502415459 20200430 3430 893 2537 26.034985422740526 73.96501457725948 20200429 3144 736 2408 23.40966921119593 76.59033078880407 20200428 2809 626 2183 22.285510857956567 77.71448914204343 20200427 8311 906 7405 10.901215256888461 89.09878474311154 20200426 7542 815 6727 10.806152214266774 89.19384778573323 20200425 4407 1150 3257 26.094849103698657 73.90515089630134 20200424 4616 879 3737 19.042461005199307 80.9575389948007 20200423 3571 962 2609 26.939232707924948 73.06076729207504 20200422 2894 582 2312 20.11057360055287 79.88942639944713 20200421 2238 509 1729 22.743521000893654 77.25647899910635 20200420 3506 854 2652 24.358243011979464 75.64175698802053 20200419 2521 522 1999 20.706069020230068 79.29393097976994 20200418 3361 736 2625 21.89824457006843 78.10175542993156 20200417 3166 788 2378 24.88945041061276 75.11054958938725 20200416 3080 752 2328 24.415584415584416 75.58441558441558 20200415 2030 560 1470 27.586206896551722 72.41379310344827 20200414 1982 536 1446 27.04339051463169 72.95660948536832 20200413 1987 711 1276 35.782586814292905 64.2174131857071 20200412 2526 531 1995 21.021377672209027 78.97862232779097 20200411 2790 726 2064 26.021505376344084 73.9784946236559 20200410 2919 783 2136 26.82425488180884 73.17574511819116 20200409 3067 656 2411 21.388979458754484 78.61102054124551 20200408 6835 1158 5677 16.942209217264082 83.05779078273592 20200407 2010 326 1684 16.218905472636816 83.78109452736318 20200406 1280 436 844 34.0625 65.9375 20200405 2727 484 2243 17.74844151081775 82.25155848918226 20200404 1920 367 1553 19.114583333333332 80.88541666666667 20200403 2469 427 2042 17.294451194815714 82.70554880518428 20200402 2003 346 1657 17.274088866699948 82.72591113330004 20200401 2690 325 2365 12.0817843866171 87.9182156133829 20200331 1799 247 1552 13.729849916620346 86.27015008337966 20200330 1136 174 962 15.316901408450704 84.6830985915493 20200329 1085 247 838 22.764976958525345 77

Percentage of positive cases reported - as per the article quoted in my blog post it looks like it's going down recently, about a half or even a one third of what is was in the higher months (April/May)

In [24]:
%sql
Select date,totalTestResultsIncrease,positiveIncrease,negativeIncrease, 
positiveIncrease/totalTestResultsIncrease*100 as PercentPositive, negativeIncrease/totalTestResultsIncrease*100 as PercentNegative
from vw_SQLMDCovid19
order by date

date totalTestResultsIncrease positiveIncrease negativeIncrease PercentPositive PercentNegative 20200329 1085 247 838 22.764976958525345 77.23502304147466 20200330 1136 174 962 15.316901408450704 84.6830985915493 20200331 1799 247 1552 13.729849916620346 86.27015008337966 20200401 2690 325 2365 12.0817843866171 87.9182156133829 20200402 2003 346 1657 17.274088866699948 82.72591113330004 20200403 2469 427 2042 17.294451194815714 82.70554880518428 20200404 1920 367 1553 19.114583333333332 80.88541666666667 20200405 2727 484 2243 17.74844151081775 82.25155848918226 20200406 1280 436 844 34.0625 65.9375 20200407 2010 326 1684 16.218905472636816 83.78109452736318 20200408 6835 1158 5677 16.942209217264082 83.05779078273592 20200409 3067 656 2411 21.388979458754484 78.61102054124551 20200410 2919 783 2136 26.82425488180884 73.17574511819116 20200411 2790 726 2064 26.021505376344084 73.9784946236559 20200412 2526 531 1995 21.021377672209027 78.97862232779097 20200413 1987 711 1276 35.782586814292905 64.2174131857071 20200414 1982 536 1446 27.04339051463169 72.95660948536832 20200415 2030 560 1470 27.586206896551722 72.41379310344827 20200416 3080 752 2328 24.415584415584416 75.58441558441558 20200417 3166 788 2378 24.88945041061276 75.11054958938725 20200418 3361 736 2625 21.89824457006843 78.10175542993156 20200419 2521 522 1999 20.706069020230068 79.29393097976994 20200420 3506 854 2652 24.358243011979464 75.64175698802053 20200421 2238 509 1729 22.743521000893654 77.25647899910635 20200422 2894 582 2312 20.11057360055287 79.88942639944713 20200423 3571 962 2609 26.939232707924948 73.06076729207504 20200424 4616 879 3737 19.042461005199307 80.9575389948007 20200425 4407 1150 3257 26.094849103698657 73.90515089630134 20200426 7542 815 6727 10.806152214266774 89.19384778573323 20200427 8311 906 7405 10.901215256888461 89.09878474311154 20200428 2809 626 2183 22.285510857956567 77.71448914204343 20200429 3144 736 2408 23.40966921119593 76.59033078880407 20200430 3430 893 2537 26.034985422740526 73.96501457725948 20200501 6624 1730 4894 26.117149758454104 73.8828502415459 20200502 4539 1001 3538 22.053315708305792 77.9466842916942 20200503 7272 989 6283 13.600110011001101 86.39988998899891 20200504 4201 946 3255 22.51844798857415 77.48155201142585 20200505 3108 709 2399 22.81209781209781 77.1879021879022 20200506 3909 1046 2863 26.7587618316705 73.24123816832949 20200507 4588 1211 3377 26.3949433304272 73.6050566695728 20200508 3587 1111 2476 30.972957903540564 69.02704209645944 20200509 3841 1049 2792 27.310596198906534 72.68940380109346 20200510 3903 1053 2850 26.979246733282093 73.0207532667179 20200511 4849 786 4063 16.209527737677874 83.79047226232213 20200512 4723 688 4035 14.56701249206013 85.43298750793987 20200513 4071 751 3320 18.447555883075413 81.55244411692459 20200514 4880 1091 3789 22.356557377049178 77.64344262295081 20200515 4372 1083 3289 24.771271729185727 75.22872827081427 20200516 7349 982 6367 13.362362226153218 86.63763777384679 20200517 4751 836 3915 17.596295516733317 82.40370448326668 20200518 6580 958 5622 14.559270516717326 85.44072948328267 20200519 7152 1784 5368 24.94407158836689 75.0559284116331 20200520 6672 777 5895 11.64568345323741 88.35431654676259 20200521 4903 1208 3695 24.637976748929226 75.36202325107078 20200522 7669 893 6776 11.64428217499022 88.35571782500978 20200523 4425 1071 3354 24.203389830508474 75.79661016949153 20200524 8035 818 7217 10.180460485376477 89.81953951462353 20200525 9215 839 8376 9.104720564297342 90.89527943570266 20200526 4910 535 4375 10.896130346232178 89.10386965376782 20200527 7568 736 6832 9.725158562367865 90.27484143763213 20200528 12803 1286 11517 10.044520815433883 89.95547918456612 20200529 9660 1279 8381 13.24016563146998 86.75983436853002 20200530 9428 1027 8401 10.893084429359355 89.10691557064064 20200531 7935 763 7172 9.615626969124133 90.38437303087586 20200601 6849 549 6300 8.015768725361367 91.98423127463863 20200602 7085 848 6237 11.968948482709951 88

The chart below shows a different story.  This is the number of cases reported per day, against the number of total tests done for the particular day. It is a numbers to numbers comparison; unlike the comparison made on Maryland's official website which shows a percentage of positive cases against number of total tests administered.

As you can see, over time, the state received more and more test results and the percentage (above) of positive cases has gone down.  This is not an indication of cases going down however.  If you think about it, just because you had a lower percentage, doesn't necessarily mean that cases are going down; your test population was lower.  Unless you test everyone everyday, it's like comparing apples to oranges; yes they are both fruit but each is still different than the other.

For example, if you look at the raw data you can start to see how statistics can tell a different story based on how you use them:
(Based on Data from June 2, 2020 run)

April 13, 2020 - # of Positive Cases 711 vs Percentage of Positive Cases ~36%

May 19, 2020 - # of Positive Cases 1784 vs Percentage of Positive Cases ~25%

May 28, 2020 - # of Positive Cases 1286 vs Percentage of Positive Cases ~10%

Are there really 10% fewer cases between April 3th and May 19th?

Was there a drop of 72% in positive cases between April 13 and May 28th?

It is very hard to determine that just based on percentages, because we don't have a equal sample of test cases on both days.  Plus how many of those recovered, how many of those got reinfect and retested.  Who knows???

In [26]:
%sql
Select date,totalTestResultsIncrease,positiveIncrease,negativeIncrease, 
positiveIncrease/totalTestResultsIncrease*100 as PercentPositive, negativeIncrease/totalTestResultsIncrease*100 as PercentNegative
from vw_SQLMDCovid19
order by date

date totalTestResultsIncrease positiveIncrease negativeIncrease PercentPositive PercentNegative 20200329 1085 247 838 22.764976958525345 77.23502304147466 20200330 1136 174 962 15.316901408450704 84.6830985915493 20200331 1799 247 1552 13.729849916620346 86.27015008337966 20200401 2690 325 2365 12.0817843866171 87.9182156133829 20200402 2003 346 1657 17.274088866699948 82.72591113330004 20200403 2469 427 2042 17.294451194815714 82.70554880518428 20200404 1920 367 1553 19.114583333333332 80.88541666666667 20200405 2727 484 2243 17.74844151081775 82.25155848918226 20200406 1280 436 844 34.0625 65.9375 20200407 2010 326 1684 16.218905472636816 83.78109452736318 20200408 6835 1158 5677 16.942209217264082 83.05779078273592 20200409 3067 656 2411 21.388979458754484 78.61102054124551 20200410 2919 783 2136 26.82425488180884 73.17574511819116 20200411 2790 726 2064 26.021505376344084 73.9784946236559 20200412 2526 531 1995 21.021377672209027 78.97862232779097 20200413 1987 711 1276 35.782586814292905 64.2174131857071 20200414 1982 536 1446 27.04339051463169 72.95660948536832 20200415 2030 560 1470 27.586206896551722 72.41379310344827 20200416 3080 752 2328 24.415584415584416 75.58441558441558 20200417 3166 788 2378 24.88945041061276 75.11054958938725 20200418 3361 736 2625 21.89824457006843 78.10175542993156 20200419 2521 522 1999 20.706069020230068 79.29393097976994 20200420 3506 854 2652 24.358243011979464 75.64175698802053 20200421 2238 509 1729 22.743521000893654 77.25647899910635 20200422 2894 582 2312 20.11057360055287 79.88942639944713 20200423 3571 962 2609 26.939232707924948 73.06076729207504 20200424 4616 879 3737 19.042461005199307 80.9575389948007 20200425 4407 1150 3257 26.094849103698657 73.90515089630134 20200426 7542 815 6727 10.806152214266774 89.19384778573323 20200427 8311 906 7405 10.901215256888461 89.09878474311154 20200428 2809 626 2183 22.285510857956567 77.71448914204343 20200429 3144 736 2408 23.40966921119593 76.59033078880407 20200430 3430 893 2537 26.034985422740526 73.96501457725948 20200501 6624 1730 4894 26.117149758454104 73.8828502415459 20200502 4539 1001 3538 22.053315708305792 77.9466842916942 20200503 7272 989 6283 13.600110011001101 86.39988998899891 20200504 4201 946 3255 22.51844798857415 77.48155201142585 20200505 3108 709 2399 22.81209781209781 77.1879021879022 20200506 3909 1046 2863 26.7587618316705 73.24123816832949 20200507 4588 1211 3377 26.3949433304272 73.6050566695728 20200508 3587 1111 2476 30.972957903540564 69.02704209645944 20200509 3841 1049 2792 27.310596198906534 72.68940380109346 20200510 3903 1053 2850 26.979246733282093 73.0207532667179 20200511 4849 786 4063 16.209527737677874 83.79047226232213 20200512 4723 688 4035 14.56701249206013 85.43298750793987 20200513 4071 751 3320 18.447555883075413 81.55244411692459 20200514 4880 1091 3789 22.356557377049178 77.64344262295081 20200515 4372 1083 3289 24.771271729185727 75.22872827081427 20200516 7349 982 6367 13.362362226153218 86.63763777384679 20200517 4751 836 3915 17.596295516733317 82.40370448326668 20200518 6580 958 5622 14.559270516717326 85.44072948328267 20200519 7152 1784 5368 24.94407158836689 75.0559284116331 20200520 6672 777 5895 11.64568345323741 88.35431654676259 20200521 4903 1208 3695 24.637976748929226 75.36202325107078 20200522 7669 893 6776 11.64428217499022 88.35571782500978 20200523 4425 1071 3354 24.203389830508474 75.79661016949153 20200524 8035 818 7217 10.180460485376477 89.81953951462353 20200525 9215 839 8376 9.104720564297342 90.89527943570266 20200526 4910 535 4375 10.896130346232178 89.10386965376782 20200527 7568 736 6832 9.725158562367865 90.27484143763213 20200528 12803 1286 11517 10.044520815433883 89.95547918456612 20200529 9660 1279 8381 13.24016563146998 86.75983436853002 20200530 9428 1027 8401 10.893084429359355 89.10691557064064 20200531 7935 763 7172 9.615626969124133 90.38437303087586 20200601 6849 549 6300 8.015768725361367 91.98423127463863 20200602 7085 848 6237 11.968948482709951 88

In [27]:
%sql
Select date,totalTestResultsIncrease,positiveIncrease,negativeIncrease, 
positiveIncrease/totalTestResultsIncrease*100 as PercentPositive, negativeIncrease/totalTestResultsIncrease*100 as PercentNegative
from vw_SQLMDCovid19
order by date

date totalTestResultsIncrease positiveIncrease negativeIncrease PercentPositive PercentNegative 20200329 1085 247 838 22.764976958525345 77.23502304147466 20200330 1136 174 962 15.316901408450704 84.6830985915493 20200331 1799 247 1552 13.729849916620346 86.27015008337966 20200401 2690 325 2365 12.0817843866171 87.9182156133829 20200402 2003 346 1657 17.274088866699948 82.72591113330004 20200403 2469 427 2042 17.294451194815714 82.70554880518428 20200404 1920 367 1553 19.114583333333332 80.88541666666667 20200405 2727 484 2243 17.74844151081775 82.25155848918226 20200406 1280 436 844 34.0625 65.9375 20200407 2010 326 1684 16.218905472636816 83.78109452736318 20200408 6835 1158 5677 16.942209217264082 83.05779078273592 20200409 3067 656 2411 21.388979458754484 78.61102054124551 20200410 2919 783 2136 26.82425488180884 73.17574511819116 20200411 2790 726 2064 26.021505376344084 73.9784946236559 20200412 2526 531 1995 21.021377672209027 78.97862232779097 20200413 1987 711 1276 35.782586814292905 64.2174131857071 20200414 1982 536 1446 27.04339051463169 72.95660948536832 20200415 2030 560 1470 27.586206896551722 72.41379310344827 20200416 3080 752 2328 24.415584415584416 75.58441558441558 20200417 3166 788 2378 24.88945041061276 75.11054958938725 20200418 3361 736 2625 21.89824457006843 78.10175542993156 20200419 2521 522 1999 20.706069020230068 79.29393097976994 20200420 3506 854 2652 24.358243011979464 75.64175698802053 20200421 2238 509 1729 22.743521000893654 77.25647899910635 20200422 2894 582 2312 20.11057360055287 79.88942639944713 20200423 3571 962 2609 26.939232707924948 73.06076729207504 20200424 4616 879 3737 19.042461005199307 80.9575389948007 20200425 4407 1150 3257 26.094849103698657 73.90515089630134 20200426 7542 815 6727 10.806152214266774 89.19384778573323 20200427 8311 906 7405 10.901215256888461 89.09878474311154 20200428 2809 626 2183 22.285510857956567 77.71448914204343 20200429 3144 736 2408 23.40966921119593 76.59033078880407 20200430 3430 893 2537 26.034985422740526 73.96501457725948 20200501 6624 1730 4894 26.117149758454104 73.8828502415459 20200502 4539 1001 3538 22.053315708305792 77.9466842916942 20200503 7272 989 6283 13.600110011001101 86.39988998899891 20200504 4201 946 3255 22.51844798857415 77.48155201142585 20200505 3108 709 2399 22.81209781209781 77.1879021879022 20200506 3909 1046 2863 26.7587618316705 73.24123816832949 20200507 4588 1211 3377 26.3949433304272 73.6050566695728 20200508 3587 1111 2476 30.972957903540564 69.02704209645944 20200509 3841 1049 2792 27.310596198906534 72.68940380109346 20200510 3903 1053 2850 26.979246733282093 73.0207532667179 20200511 4849 786 4063 16.209527737677874 83.79047226232213 20200512 4723 688 4035 14.56701249206013 85.43298750793987 20200513 4071 751 3320 18.447555883075413 81.55244411692459 20200514 4880 1091 3789 22.356557377049178 77.64344262295081 20200515 4372 1083 3289 24.771271729185727 75.22872827081427 20200516 7349 982 6367 13.362362226153218 86.63763777384679 20200517 4751 836 3915 17.596295516733317 82.40370448326668 20200518 6580 958 5622 14.559270516717326 85.44072948328267 20200519 7152 1784 5368 24.94407158836689 75.0559284116331 20200520 6672 777 5895 11.64568345323741 88.35431654676259 20200521 4903 1208 3695 24.637976748929226 75.36202325107078 20200522 7669 893 6776 11.64428217499022 88.35571782500978 20200523 4425 1071 3354 24.203389830508474 75.79661016949153 20200524 8035 818 7217 10.180460485376477 89.81953951462353 20200525 9215 839 8376 9.104720564297342 90.89527943570266 20200526 4910 535 4375 10.896130346232178 89.10386965376782 20200527 7568 736 6832 9.725158562367865 90.27484143763213 20200528 12803 1286 11517 10.044520815433883 89.95547918456612 20200529 9660 1279 8381 13.24016563146998 86.75983436853002 20200530 9428 1027 8401 10.893084429359355 89.10691557064064 20200531 7935 763 7172 9.615626969124133 90.38437303087586 20200601 6849 549 6300 8.015768725361367 91.98423127463863 20200602 7085 848 6237 11.968948482709951 88

In [28]:
%sql
Select date,totalTestResultsIncrease,positiveIncrease,negativeIncrease, 
positiveIncrease/totalTestResultsIncrease*100 as PercentPositive, negativeIncrease/totalTestResultsIncrease*100 as PercentNegative
from vw_SQLMDCovid19
order by date

date totalTestResultsIncrease positiveIncrease negativeIncrease PercentPositive PercentNegative 20200329 1085 247 838 22.764976958525345 77.23502304147466 20200330 1136 174 962 15.316901408450704 84.6830985915493 20200331 1799 247 1552 13.729849916620346 86.27015008337966 20200401 2690 325 2365 12.0817843866171 87.9182156133829 20200402 2003 346 1657 17.274088866699948 82.72591113330004 20200403 2469 427 2042 17.294451194815714 82.70554880518428 20200404 1920 367 1553 19.114583333333332 80.88541666666667 20200405 2727 484 2243 17.74844151081775 82.25155848918226 20200406 1280 436 844 34.0625 65.9375 20200407 2010 326 1684 16.218905472636816 83.78109452736318 20200408 6835 1158 5677 16.942209217264082 83.05779078273592 20200409 3067 656 2411 21.388979458754484 78.61102054124551 20200410 2919 783 2136 26.82425488180884 73.17574511819116 20200411 2790 726 2064 26.021505376344084 73.9784946236559 20200412 2526 531 1995 21.021377672209027 78.97862232779097 20200413 1987 711 1276 35.782586814292905 64.2174131857071 20200414 1982 536 1446 27.04339051463169 72.95660948536832 20200415 2030 560 1470 27.586206896551722 72.41379310344827 20200416 3080 752 2328 24.415584415584416 75.58441558441558 20200417 3166 788 2378 24.88945041061276 75.11054958938725 20200418 3361 736 2625 21.89824457006843 78.10175542993156 20200419 2521 522 1999 20.706069020230068 79.29393097976994 20200420 3506 854 2652 24.358243011979464 75.64175698802053 20200421 2238 509 1729 22.743521000893654 77.25647899910635 20200422 2894 582 2312 20.11057360055287 79.88942639944713 20200423 3571 962 2609 26.939232707924948 73.06076729207504 20200424 4616 879 3737 19.042461005199307 80.9575389948007 20200425 4407 1150 3257 26.094849103698657 73.90515089630134 20200426 7542 815 6727 10.806152214266774 89.19384778573323 20200427 8311 906 7405 10.901215256888461 89.09878474311154 20200428 2809 626 2183 22.285510857956567 77.71448914204343 20200429 3144 736 2408 23.40966921119593 76.59033078880407 20200430 3430 893 2537 26.034985422740526 73.96501457725948 20200501 6624 1730 4894 26.117149758454104 73.8828502415459 20200502 4539 1001 3538 22.053315708305792 77.9466842916942 20200503 7272 989 6283 13.600110011001101 86.39988998899891 20200504 4201 946 3255 22.51844798857415 77.48155201142585 20200505 3108 709 2399 22.81209781209781 77.1879021879022 20200506 3909 1046 2863 26.7587618316705 73.24123816832949 20200507 4588 1211 3377 26.3949433304272 73.6050566695728 20200508 3587 1111 2476 30.972957903540564 69.02704209645944 20200509 3841 1049 2792 27.310596198906534 72.68940380109346 20200510 3903 1053 2850 26.979246733282093 73.0207532667179 20200511 4849 786 4063 16.209527737677874 83.79047226232213 20200512 4723 688 4035 14.56701249206013 85.43298750793987 20200513 4071 751 3320 18.447555883075413 81.55244411692459 20200514 4880 1091 3789 22.356557377049178 77.64344262295081 20200515 4372 1083 3289 24.771271729185727 75.22872827081427 20200516 7349 982 6367 13.362362226153218 86.63763777384679 20200517 4751 836 3915 17.596295516733317 82.40370448326668 20200518 6580 958 5622 14.559270516717326 85.44072948328267 20200519 7152 1784 5368 24.94407158836689 75.0559284116331 20200520 6672 777 5895 11.64568345323741 88.35431654676259 20200521 4903 1208 3695 24.637976748929226 75.36202325107078 20200522 7669 893 6776 11.64428217499022 88.35571782500978 20200523 4425 1071 3354 24.203389830508474 75.79661016949153 20200524 8035 818 7217 10.180460485376477 89.81953951462353 20200525 9215 839 8376 9.104720564297342 90.89527943570266 20200526 4910 535 4375 10.896130346232178 89.10386965376782 20200527 7568 736 6832 9.725158562367865 90.27484143763213 20200528 12803 1286 11517 10.044520815433883 89.95547918456612 20200529 9660 1279 8381 13.24016563146998 86.75983436853002 20200530 9428 1027 8401 10.893084429359355 89.10691557064064 20200531 7935 763 7172 9.615626969124133 90.38437303087586 20200601 6849 549 6300 8.015768725361367 91.98423127463863 20200602 7085 848 6237 11.968948482709951 88

Look at this chart, this is a percentage of positive vs negative cases (percent) but it is over the TOTAL aggregate over time (running total).

It shows that for the last two months, positive cases have been around the 17-20% range. If you go back to March, that range gets wider to 9-20%.

Again, another way to tell a different story with the same data.

I love this bar graph, because it shows that although Negative results are increasing, Positive are not really decreasing but they are increasing also.

Just look at the raw numbers.

April 13, 2020 - 8936 total positive (cumulative)  - 42815 total negative (cumulative)  - total cumulative 51751 - percent positive ~17.27%

June 1, 2020 - 53327 total positive (cumulative) - 255403 total negative (cumulative) - total cumulative 308730 - percent positive ~17.27%

In [31]:
%sql
Select date,totalTestResults,positive,negative,
positive/totalTestResults*100 as PercentPositive, negative/totalTestResults*100 as PercentNegative
from vw_SQLMDCovid19
order by date

date totalTestResults positive negative PercentPositive PercentNegative 20200329 13593 1239 12354 9.114985654380932 90.88501434561907 20200330 14729 1413 13316 9.593319302057166 90.40668069794283 20200331 16528 1660 14868 10.043562439496611 89.95643756050339 20200401 19218 1985 17233 10.328858361952337 89.67114163804767 20200402 21221 2331 18890 10.984402243061119 89.01559775693887 20200403 23690 2758 20932 11.642043056141832 88.35795694385816 20200404 25610 3125 22485 12.202264740335806 87.7977352596642 20200405 28337 3609 24728 12.735998870734374 87.26400112926562 20200406 29617 4045 25572 13.65769659317284 86.34230340682716 20200407 31627 4371 27256 13.820469851708983 86.17953014829102 20200408 38462 5529 32933 14.375227497270032 85.62477250272997 20200409 41529 6185 35344 14.893207156444893 85.1067928435551 20200410 44448 6968 37480 15.676745860331174 84.32325413966882 20200411 47238 7694 39544 16.287734451077522 83.71226554892249 20200412 49764 8225 41539 16.52801221766739 83.47198778233262 20200413 51751 8936 42815 17.26729918262449 82.73270081737552 20200414 53733 9472 44261 17.62790091749949 82.37209908250051 20200415 55763 10032 45731 17.990423757688788 82.00957624231121 20200416 58843 10784 48059 18.326733851095288 81.67326614890472 20200417 62009 11572 50437 18.661807157025592 81.33819284297441 20200418 65370 12308 53062 18.828208658405995 81.171791341594 20200419 67891 12830 55061 18.897939343948387 81.1020606560516 20200420 71397 13684 57713 19.166071403560373 80.83392859643963 20200421 73635 14193 59442 19.27480138521084 80.72519861478916 20200422 76529 14775 61754 19.306406721634936 80.69359327836506 20200423 80100 15737 64363 19.646691635455678 80.35330836454432 20200424 84716 16616 68100 19.61376835544643 80.38623164455356 20200425 89123 17766 71357 19.934248173872067 80.06575182612794 20200426 96665 18581 78084 19.222055552681944 80.77794444731806 20200427 104976 19487 85489 18.563290656912056 81.43670934308794 20200428 107785 20113 87672 18.660295959549103 81.3397040404509 20200429 110929 20849 90080 18.79490484904759 81.20509515095242 20200430 114359 21742 92617 19.01205851747567 80.98794148252433 20200501 120983 23472 97511 19.401072878007657 80.59892712199235 20200502 125522 24473 101049 19.496980608976912 80.50301939102309 20200503 132794 25462 107332 19.17405906893384 80.82594093106617 20200504 136995 26408 110587 19.276615934888135 80.72338406511187 20200505 140103 27117 112986 19.355045930493993 80.644954069506 20200506 144012 28163 115849 19.556009221453767 80.44399077854624 20200507 148600 29374 119226 19.7671601615074 80.2328398384926 20200508 152187 30485 121702 20.031277310151328 79.96872268984868 20200509 156028 31534 124494 20.210475042941013 79.78952495705899 20200510 159931 32587 127344 20.375662004239327 79.62433799576067 20200511 164780 33373 131407 20.25306469231703 79.74693530768297 20200512 169503 34061 135442 20.094629593576517 79.90537040642349 20200513 173574 34812 138762 20.05599917038266 79.94400082961734 20200514 178454 35903 142551 20.11891019534446 79.88108980465553 20200515 182826 36986 145840 20.230164199840285 79.76983580015971 20200516 190175 37968 152207 19.964769291442092 80.0352307085579 20200517 194926 38804 156122 19.90704164657357 80.09295835342643 20200518 201506 39762 161744 19.732414915684892 80.2675850843151 20200519 208658 41546 167112 19.91105061871579 80.08894938128421 20200520 215330 42323 173007 19.654948219012677 80.34505178098732 20200521 220233 43531 176702 19.765884313431684 80.23411568656832 20200522 227902 44424 183478 19.492588919798862 80.50741108020114 20200523 232327 45495 186832 19.582312860752303 80.41768713924769 20200524 240362 46313 194049 19.268020735390788 80.73197926460921 20200525 249577 47152 202425 18.892766561021247 81.10723343897875 20200526 254487 47687 206800 18.738481729911545 81.26151827008844 20200527 262055 48423 213632 18.478182061017726 81.52181793898228 20200528 274858 49709 225149 18.08533861121015 81.91466138878984 20200529 284

Like the chart above, this one gives you the percentage of Positive from the cumulative number of tests and the same percentage for negative. Pretty much what I explained above.  You can see from the graph, there has not been much change.

In [33]:
%sql
Select date,totalTestResults,positive,negative, 
positive/totalTestResults*100 as PercentPositive, negative/totalTestResults*100 as PercentNegative
from vw_SQLMDCovid19
order by date

date totalTestResults positive negative PercentPositive PercentNegative 20200329 13593 1239 12354 9.114985654380932 90.88501434561907 20200330 14729 1413 13316 9.593319302057166 90.40668069794283 20200331 16528 1660 14868 10.043562439496611 89.95643756050339 20200401 19218 1985 17233 10.328858361952337 89.67114163804767 20200402 21221 2331 18890 10.984402243061119 89.01559775693887 20200403 23690 2758 20932 11.642043056141832 88.35795694385816 20200404 25610 3125 22485 12.202264740335806 87.7977352596642 20200405 28337 3609 24728 12.735998870734374 87.26400112926562 20200406 29617 4045 25572 13.65769659317284 86.34230340682716 20200407 31627 4371 27256 13.820469851708983 86.17953014829102 20200408 38462 5529 32933 14.375227497270032 85.62477250272997 20200409 41529 6185 35344 14.893207156444893 85.1067928435551 20200410 44448 6968 37480 15.676745860331174 84.32325413966882 20200411 47238 7694 39544 16.287734451077522 83.71226554892249 20200412 49764 8225 41539 16.52801221766739 83.47198778233262 20200413 51751 8936 42815 17.26729918262449 82.73270081737552 20200414 53733 9472 44261 17.62790091749949 82.37209908250051 20200415 55763 10032 45731 17.990423757688788 82.00957624231121 20200416 58843 10784 48059 18.326733851095288 81.67326614890472 20200417 62009 11572 50437 18.661807157025592 81.33819284297441 20200418 65370 12308 53062 18.828208658405995 81.171791341594 20200419 67891 12830 55061 18.897939343948387 81.1020606560516 20200420 71397 13684 57713 19.166071403560373 80.83392859643963 20200421 73635 14193 59442 19.27480138521084 80.72519861478916 20200422 76529 14775 61754 19.306406721634936 80.69359327836506 20200423 80100 15737 64363 19.646691635455678 80.35330836454432 20200424 84716 16616 68100 19.61376835544643 80.38623164455356 20200425 89123 17766 71357 19.934248173872067 80.06575182612794 20200426 96665 18581 78084 19.222055552681944 80.77794444731806 20200427 104976 19487 85489 18.563290656912056 81.43670934308794 20200428 107785 20113 87672 18.660295959549103 81.3397040404509 20200429 110929 20849 90080 18.79490484904759 81.20509515095242 20200430 114359 21742 92617 19.01205851747567 80.98794148252433 20200501 120983 23472 97511 19.401072878007657 80.59892712199235 20200502 125522 24473 101049 19.496980608976912 80.50301939102309 20200503 132794 25462 107332 19.17405906893384 80.82594093106617 20200504 136995 26408 110587 19.276615934888135 80.72338406511187 20200505 140103 27117 112986 19.355045930493993 80.644954069506 20200506 144012 28163 115849 19.556009221453767 80.44399077854624 20200507 148600 29374 119226 19.7671601615074 80.2328398384926 20200508 152187 30485 121702 20.031277310151328 79.96872268984868 20200509 156028 31534 124494 20.210475042941013 79.78952495705899 20200510 159931 32587 127344 20.375662004239327 79.62433799576067 20200511 164780 33373 131407 20.25306469231703 79.74693530768297 20200512 169503 34061 135442 20.094629593576517 79.90537040642349 20200513 173574 34812 138762 20.05599917038266 79.94400082961734 20200514 178454 35903 142551 20.11891019534446 79.88108980465553 20200515 182826 36986 145840 20.230164199840285 79.76983580015971 20200516 190175 37968 152207 19.964769291442092 80.0352307085579 20200517 194926 38804 156122 19.90704164657357 80.09295835342643 20200518 201506 39762 161744 19.732414915684892 80.2675850843151 20200519 208658 41546 167112 19.91105061871579 80.08894938128421 20200520 215330 42323 173007 19.654948219012677 80.34505178098732 20200521 220233 43531 176702 19.765884313431684 80.23411568656832 20200522 227902 44424 183478 19.492588919798862 80.50741108020114 20200523 232327 45495 186832 19.582312860752303 80.41768713924769 20200524 240362 46313 194049 19.268020735390788 80.73197926460921 20200525 249577 47152 202425 18.892766561021247 81.10723343897875 20200526 254487 47687 206800 18.738481729911545 81.26151827008844 20200527 262055 48423 213632 18.478182061017726 81.52181793898228 20200528 274858 49709 225149 18.08533861121015 81.91466138878984 20200529 284